src/ucs/sys/topo.h

```diff
@@ -0,0 +1,64 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_TOPO_H
+#define UCS_TOPO_H
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucs/type/status.h>
+#include <limits.h>
+
+BEGIN_C_DECLS
+
+/** @file topo.h */
+
+/**
+ * System device type
+ */
+typedef enum {
+    UCS_SYS_DEVICE_IB = 0, /* Infiniband Device */
+    UCS_SYS_DEVICE_CUDA,   /* NVIDIA GPU Device */
```

Comment:
not sure why we need this enum
seems too specific

Reply:
Agreed. Actually not needed. At UCP-level when we make {rma/am}_bw_lanes per numa_node/gpu_memory_node, the main result of interest is in building a priority list of transports based on distance between two system devices -- one conerning the memory node and one the transport device. I guess the internal implementation of getting inter-device distance will worry about what kind of device the structure is associated with (if at all). 

Will remove the enum.

Reply:
BTW, i would cleanup this PR to have API only. and also define the API for distance between 2 devices in terms of latency and BW. The implementation could be stub.
maybe add function like ucs_topo_print() to be used by ucx_info to draw some topology map.

Reply:
+1

I'll create a PR based on the latest master with the above suggested changes.

Reply:
@yosefe I've created a new PR here: https://github.com/openucx/ucx/pull/4386

Plan is to create populate a list of devices discovered on the system and use [`ucs_topo_get_distance()`](https://github.com/openucx/ucx/pull/4386/files#diff-cc86b76236551c380fe37b8b50a3262dR89) to implement [`ucs_topo_print()`](https://github.com/openucx/ucx/pull/4386/files#diff-cc86b76236551c380fe37b8b50a3262dR96) as a follow up to https://github.com/openucx/ucx/pull/4386